apps/web/src/services/asset-inventory/collector/collector-detail/modules/CollectorServiceAccountsSection.vue

```diff
@@ -102,17 +102,30 @@ const fetchCollectorUpdate = async (): Promise<CollectorModel> => {
         secret_filter: {
             ...originSecretFilter,
             state: collectorFormState.attachedServiceAccountType === 'specific' ? 'ENABLED' : 'DISABLED',
-            service_accounts: collectorFormState.attachedServiceAccountType === 'specific'
-                ? collectorFormStore.serviceAccounts
-                : [],
         },
     };
+    const serviceAccountParams = collectorFormState.selectedServiceAccountFilterOptions === 'include' ? {
+        service_accounts: collectorFormStore.serviceAccounts,
+    } : {
+        exclude_service_accounts: collectorFormStore.serviceAccounts,
+    };
+    Object.assign(params.secret_filter ?? {}, serviceAccountParams);
     return SpaceConnector.client.inventory.collector.update(params);
 };
 
 
 /* event handlers */
 const handleClickEdit = () => {
+    const isExcludeOption = !!collectorFormState.originCollector?.secret_filter?.exclude_service_accounts?.length;
+    if (isExcludeOption) {
+        collectorFormStore.$patch({
+            selectedServiceAccountFilterOptions: 'exclude',
```

Comment:
nitpick: What about to use singular instead of plural? `selectedServiceAccountFilterOption`

Reply:
Wow, thank you for checking.